File: src/main/java/com/bolero/bootcamp/Bootcamp/entity/Department.java
package com.bolero.bootcamp.Bootcamp.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Department {
    @Id
    @GeneratedValue(strategy=GenerationType.SEQUENCE, generator = "DEPARTMENT_SEQ_GEN")
    @SequenceGenerator(name = "DEPARTMENT_SEQ_GEN", sequenceName = "SEQ_DEPARTMENT_ID", allocationSize = 1)
    private Long id;

    @Column(nullable = false, unique = true)
    private String name;

    @Column(nullable = false)
    private Boolean mandatory;

    @Column(nullable = false)
    private Boolean readOnly;
}

File: userInterface/src/app/models/department.ts
// department.ts
export class Department {
  id: number; // Required property
  name: string;
  mandatory: boolean;
  readOnly: boolean;

  constructor(id: number, name: string, mandatory: boolean, readOnly: boolean) {
    this.id = id;
    this.name = name;
    this.mandatory = mandatory;
    this.readOnly = readOnly;
  }
}
